src/dune_rules/mdx.ml

```diff
@@ -68,19 +68,46 @@ module Deps = struct
       [ A "deps"; Dep (Path.build files.Files.src) ]
       ~stdout_to:files.Files.deps
 
-  let to_path ~dir str =
-    let local = Path.Local.of_string str in
-    let build = Path.Build.append_local dir local in
+  let path_escapes_dir str =
+    try
+      let (_ : Path.Local.t) = Path.Local.of_string str in
```

Comment:
We don't want to catch things like code_error. Can we add an option/returning conversion and use that?

Reply:
I pushed something to add a new function. It's a bit more verbose that what I expected but this seems to do the trick. A more conservative fix is to just catch `User_error.E _` here otherwise (but I agree it's not great). What do you think?

Reply:
Both are acceptable.